app/server.go

```diff
@@ -868,6 +871,10 @@ func (s *Server) Shutdown() {
 		mlog.Warn("Error flushing logs", mlog.Err(err))
 	}
 
+	if s.dndTask != nil {
+		s.dndTask.Cancel()
+	}
```

Comment:
There is still a race here. Wrap it up with the mutex:
```suggestion
	s.dndnTaskMut.Lock()
	if s.dndTask != nil {
		s.dndTask.Cancel()
	}
	s.dndnTaskMut.Unlock()
```